lib/wanda/operations/reporting.ex

```diff
@@ -0,0 +1,138 @@
+defmodule Wanda.Operations.Reporting do
```

Comment:
@arbulu89 I would appreciate feedback on this approach.

Decoupling this `Reporting` module enables the mocking we want to introduce, besides improving the general code structure in terms of responsibility of components.

If you look at checks execution a similar approach made it possible in the first place to have such mocking.

Naming can be further improved: both the module name and functions name (I mainly kept similar if not the same to the ones that were embedded in the server)